app/models/watching_transaction.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'ethereum_api'
+
+class WatchingTransaction < ApplicationRecord
+  before_create :set_uuid
+
+  belongs_to :user
+
+  validates :transaction_object,
+            presence: true
+  validates :signed_transaction,
+            presence: true
+  validates :txhash,
+            presence: true,
+            uniqueness: true
+
+  def set_uuid
+    self.id = SecureRandom.uuid
+    self.group_id = id unless group_id
+  end
+
+  def transaction_object
+    JSON.parse(super)
+  end
+
+  def transaction_object=(value)
+    super(JSON.generate(value))
+  end
+
+  def txhash=(value)
+    super(value&.downcase)
+  end
+
+  class << self
+    def watch(user, attrs)
+      tx = WatchingTransaction.new(
+        user: user,
+        transaction_object: attrs.fetch(:transaction_object, nil),
+        signed_transaction: attrs.fetch(:signed_transaction, nil),
+        txhash: attrs.fetch(:txhash, nil)
+      )
+
+      return [:invalid_data, tx.errors] unless tx.valid?
+      return [:database_error, tx.errors] unless tx.save
+
+      [:ok, tx]
+    end
+
+    def resend(user, watching_transaction, attrs)
+      transaction_object = attrs.fetch(:transaction_object, nil)
+      unless user.id == watching_transaction.user.id
+        return [:unauthorized_action, nil]
+      end
+      unless transaction_object.fetch('nonce', nil) == watching_transaction.transaction_object['nonce']
+        return [:invalid_nonce, nil]
+      end
+
+      tx = WatchingTransaction.new(
+        user: user,
+        transaction_object: transaction_object,
+        signed_transaction: attrs.fetch(:signed_transaction, nil),
+        txhash: attrs.fetch(:txhash, nil),
+        group_id: watching_transaction.group_id
+      )
+
+      return [:invalid_data, tx.errors] unless tx.valid?
+      return [:database_error, tx.errors] unless tx.save
+
+      [:ok, tx]
+    end
+
+    def resend_transactions
+      WatchingTransaction.all.order('created_at ASC').each do |tx|
+        ok_tx, data = EthereumApi.get_transaction_by_hash(tx.txhash)
+        unless ok_tx == :ok && data
+          if WatchingTransaction.where(group_id: tx.group_id).count == 1
```

Comment:
This is slightly rather inefficient where every row makes a database request. Can we have a version where the only database request made is fetching the sorted transactions?